pkg/registry/client.go

```diff
@@ -14,30 +14,86 @@
 package registry
 
 import (
-	"net/http"
-	"strings"
-
-	"github.com/heroku/docker-registry-client/registry"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
 )
 
-// NewRegistryClient creates a new Registry with the given URL and credentials, then Ping()s it
-// before returning it to verify that the registry is available.
-func NewRegistryClient(registryURL, username, password string) (*registry.Registry, error) {
-	transport := http.DefaultTransport
-	return newFromTransport(registryURL, username, password, transport)
+type OktetoRegistryAuthenticator struct {
```

Comment:
Minor thing. If possible, It is really helpful (and a good practice) add comments to public exported elements (types, functions, variables, etc...). (Setting `golint` as the lint tool for Go in the VSCode helps you to detect this cases)

Reply:
If it is not needed to use our own type, maybe we can use directly the basic authenticator provided by the library: https://github.com/google/go-containerregistry/blob/main/pkg/authn/basic.go. In that way, `newBasicAuthRegistry` would just build a new `authn.Basic`

Reply:
a whole new world has been discovered for me with `golint`😍 - added some minor fixes regarding the registry package after running the linter. 

also, applied your suggestion and removed the custom type.
thnks